core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1583,6 +1583,159 @@ public static function getDriverForTable(string $table): string
+					$args[$k] = $row[$v] ? Date::parse(Config::get('dateFormat'), $row[$v]) : '-';
+				}
+				elseif (($GLOBALS['TL_DCA'][$table]['fields'][$v]['eval']['rgxp'] ?? null) == 'time')
+				{
+					$args[$k] = $row[$v] ? Date::parse(Config::get('timeFormat'), $row[$v]) : '-';
+				}
+				else
+				{
+					$args[$k] = $row[$v] ? Date::parse(Config::get('datimFormat'), $row[$v]) : '-';
+				}
+			}
+			elseif (($GLOBALS['TL_DCA'][$table]['fields'][$v]['eval']['isBoolean'] ?? null) || (($GLOBALS['TL_DCA'][$table]['fields'][$v]['inputType'] ?? null) == 'checkbox' && !($GLOBALS['TL_DCA'][$table]['fields'][$v]['eval']['multiple'] ?? null)))
+			{
+				$args[$k] = $row[$v] ? $GLOBALS['TL_LANG']['MSC']['yes'] : $GLOBALS['TL_LANG']['MSC']['no'];
+			}
+			elseif (isset($row[$v]))
+			{
+				$row_v = StringUtil::deserialize($row[$v]);
+
+				if (\is_array($row_v))
+				{
+					$args_k = array();
+
+					foreach ($row_v as $option)
+					{
+						$args_k[] = $GLOBALS['TL_DCA'][$table]['fields'][$v]['reference'][$option] ?: $option;
+					}
+
+					$args[$k] = implode(', ', $args_k);
+				}
+				elseif (isset($GLOBALS['TL_DCA'][$table]['fields'][$v]['reference'][$row[$v]]))
+				{
+					$args[$k] = \is_array($GLOBALS['TL_DCA'][$table]['fields'][$v]['reference'][$row[$v]]) ? $GLOBALS['TL_DCA'][$table]['fields'][$v]['reference'][$row[$v]][0] : $GLOBALS['TL_DCA'][$table]['fields'][$v]['reference'][$row[$v]];
+				}
+				elseif ((($GLOBALS['TL_DCA'][$table]['fields'][$v]['eval']['isAssociative'] ?? null) || ArrayUtil::isAssoc($GLOBALS['TL_DCA'][$table]['fields'][$v]['options'] ?? null)) && isset($GLOBALS['TL_DCA'][$table]['fields'][$v]['options'][$row[$v]]))
+				{
+					$args[$k] = $GLOBALS['TL_DCA'][$table]['fields'][$v]['options'][$row[$v]] ?? null;
+				}
+				else
+				{
+					$args[$k] = $row[$v];
+				}
+			}
+			else
+			{
+				$args[$k] = null;
+			}
+		}
+
+		// Render the label
+		$label = vsprintf($labelConfig['format'] ?? '%s', $args);
+
+		// Shorten the label it if it is too long
+		if (($labelConfig['maxCharacters'] ?? null) > 0 && $labelConfig['maxCharacters'] < \strlen(strip_tags($label)))
+		{
+			$label = trim(StringUtil::substrHtml($label, $labelConfig['maxCharacters'])) . ' …';
+		}
+
+		// Remove empty brackets (), [], {}, <> and empty tags from the label
+		$label = preg_replace('/\( *\) ?|\[ *] ?|{ *} ?|< *> ?/', '', $label);
+		$label = preg_replace('/<[^>]+>\s*<\/[^>]+>/', '', $label);
+
+		$mode = $GLOBALS['TL_DCA'][$table]['list']['sorting']['mode'] ?? self::MODE_SORTED;
+
+		// Execute label_callback
+		if (\is_array($labelConfig['label_callback'] ?? null) || \is_callable($labelConfig['label_callback'] ?? null))
+		{
+			if (\in_array($mode, array(self::MODE_TREE, self::MODE_TREE_EXTENDED)))
+			{
+				if (\is_array($labelConfig['label_callback'] ?? null))
+				{
+					$label = System::importStatic($labelConfig['label_callback'][0])->{$labelConfig['label_callback'][1]}($row, $label, $this, '', $protected, $isVisibleRootTrailPage);
```

Comment:
This is why the labels of the protected pages are missing: the original arguments were `$objRow->row(), $label, $this, '', false, $blnProtected, $isVisibleRootTrailPage`. It seems you have dropped the fifths argument?

Reply:
Hm, no, `$blnProtected` is passed [here](https://github.com/contao/contao/pull/3644/files#diff-e4bf1535e70e3fbc2b9692ba8add4c45237305ff815df14c7ba9c9ce53dc9204R4063) and then passed to the callback as the 5th argument in the line you referenced, but I'll analyse.

Reply:
Adding the fifth argument again fixes the issue for me:

<img width="801" alt="" src="https://user-images.githubusercontent.com/1192057/149125017-b4af4a75-4635-4f5b-ae90-32a18b1db5c0.png">

Tested in the regular view and in the picker view. No idea if I am overlooking something though.

Reply:
Aaah, yes... the argument that is always `false` (and the one before is always an empty string) 😁 I forgot about that one.